src/pykeen/losses.py

```diff
@@ -971,18 +1061,14 @@ def __init__(
         self.margin = margin
         self.margin_activation = margin_activation_resolver.make(margin_activation)
 
-    def forward(
-        self,
-        logits: FloatTensor,
-        labels: FloatTensor,
-    ) -> FloatTensor:
-        """Calculate the loss for the given scores and labels."""
-        assert 0.0 <= labels.min() and labels.max() <= 1.0
+    # docstr-coverage: inherited
+    def forward(self, x: FloatTensor, target: FloatTensor, weight: FloatTensor | None = None) -> FloatTensor:
         # scale labels from [0, 1] to [-1, 1]
-        labels = 2 * labels - 1
-        loss = self.margin_activation(self.margin - labels * logits)
-        loss = self._reduction_method(loss)
-        return loss
+        target = 2 * target - 1
+        loss = self.margin_activation(self.margin - target * x)
+        if weight is None:
+            return self._reduction_method(loss)
+        return safe_weighted_mean(loss, weight=weight)
```

Comment:
should we complain if weights are given but the reduction method is not mean?

Reply:
Done in https://github.com/pykeen/pykeen/pull/1529/commits/dfa687f2fae4108905f792abe294f67919731247 (and adding support for weighted sum, too)